src/ScopeProvider/ScopeProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import { Provider, useStore } from 'jotai/react';
+import { type ReactNode, createContext, useContext, useState } from 'react';
+import { createScope, type Scope } from './scope';
+import { AnyAtom, Store } from './types';
+
+const ScopeContext = createContext<{
+  scope: Scope | undefined;
+  baseStore: Store | undefined;
+}>({
+  scope: undefined,
+  baseStore: undefined,
+});
+
+const patchedStoreSymbol = Symbol();
+
+export function ScopeProvider({
+  atoms,
+  children,
+}: {
+  atoms: Iterable<AnyAtom>;
+  children: ReactNode;
+}) {
+  const parentStore = useStore();
+  let { scope: parentScope, baseStore = parentStore } =
+    useContext(ScopeContext);
+  if (!(patchedStoreSymbol in parentStore)) {
+    parentScope = undefined;
```

Comment:
why does parentScopeContext include storeOrUndefined? Can't you just get the store from useStore?

I realize in nested scopes useStore will return the parent scope patched store. But we really don't need and don't care about this store. We should just use whatever the base store is. Keeping the base store in state is useful for this reason.

But for the case of:
```jsx
<ScopeProvider atoms={[a]}> {/* level 1 */}
  <Provider>
    <ScopeProvider atoms={[b]}> {/* level 2 */}
      ...
```
We don't really want to consider level 2 a nested scope under level 1. To detect this case, we should mark the patched store with a special property, use useStore to get the parent store, and ignore the parent scope if the parent store does not have the patched property.

Reply:
Would you please document this case in the code? It takes me a while to figure out why the code is implemented like that.